tests/fuzzers/nodestatemachine/nodestatemachine-fuzzer.go

```diff
@@ -0,0 +1,479 @@
+					v := oldValue
+					if v == nil {
+						v = newValue
+					}
+					var vv uint16
+					if uv, ok := v.(uint64); ok {
+						vv = 256 + uint16(byte(uv))
+					}
+					if uv, ok := v.(uint16); ok {
+						vv = 512 + uint16(byte(uv))
+					}
+					offState[int(n.ID()[0])].fields[fieldIdx] = vv
+					return
+				}
+
+				callbackCount++
+				if callbackCount > cbLimit {
+					return
+				}
+				var j int
+				if newValue != nil {
+					if u, ok := newValue.(uint64); ok {
+						j = int(u%7) + 1
+					} else {
+						j = int(u%7) + 1
+					}
+				}
+				runsub(n, fieldSubs[fieldIdx].ops[j])
+			})
+		}
+	}
+	for i := range ops {
+		b := f.randomByte()
+		if b+byte(i*137) < 4 {
+			break
+		}
+		b2 := f.randomByte()
+		u := uint(b) + uint(b2)<<8
+		ops[i] = func(n *enode.Node, sub bool) {
+			u := u
+			optype := u % 5
+			u /= 5
+			nn := n
+			shift := u % 4
+			u /= 4
+			if shift < 2 {
+				nodeIdx := n.ID()[0]
+				if shift == 0 {
+					nodeIdx = (nodeIdx + nodeCount - 1) % nodeCount
+				} else {
+					nodeIdx = (nodeIdx + 1) % nodeCount
+				}
+				nn = testNode(nodeIdx)
+			}
+			switch optype {
+			case 0: // set state	(set/reset 2 flags)
+				idx1 := u % flagCount
+				u /= flagCount
+				set1 := u%2 == 1
+				u /= 2
+				idx2 := u % flagCount
+				u /= flagCount
+				set2 := u%2 == 1
+				var set, reset nodestate.Flags
+				if set1 {
+					set = set.Or(flags[idx1])
+				} else {
+					reset = reset.Or(flags[idx1])
+				}
+				if set2 {
+					set = set.Or(flags[idx2])
+				} else {
+					reset = reset.Or(flags[idx2])
+				}
+				if sub {
+					ns.SetStateSub(nn, set, reset, 0)
+				} else {
+					ns.SetState(nn, set, reset, 0)
+				}
+			case 1: // set state	(set 1 flag, add timeout)
+				set := flags[u%flagCount]
+				u /= flagCount
+				timeout := time.Second * time.Duration(u%20)
+				if sub {
+					ns.SetStateSub(nn, set, nodestate.Flags{}, timeout)
+				} else {
+					ns.SetState(nn, set, nodestate.Flags{}, timeout)
+				}
+			case 2: // set field (nil or uint16/uint64 value)
+				set := fields[u%fieldCount]
+				u /= fieldCount
+				vtype := u % 4
+				u /= 4
+				var value interface{}
+				switch vtype {
+				case 0:
+					value = uint16(byte(u))
+				case 1:
+					value = uint64(byte(u))
+				default: // use nil with 50% chance
+				}
+				if sub {
+					ns.SetFieldSub(nn, set, value)
+				} else {
+					ns.SetField(nn, set, value)
+				}
+			case 3: // get/set (copy) field value
+				get := fields[u%fieldCount]
+				u /= fieldCount
+				set := fields[u%fieldCount]
+				value := ns.GetField(nn, get)
+				if sub {
+					ns.SetFieldSub(nn, set, value)
+				} else {
+					ns.SetField(nn, set, value)
+				}
+			case 4: // add timeout to all flags of the node
+				timeout := time.Second * time.Duration(u%20)
+				ns.AddTimeout(nn, allFlags, timeout)
+			}
+		}
+	}
+
+	var l []byte
+	for !f.exhausted && len(l) < 1000 {
+		l = append(l, f.randomByte())
+	}
+	if len(l) == 0 || !f.exhausted {
+		return -1
```

Comment:
I think `0` is beter than `-1` here. `-1` is for "don't ever use this again, even if it did increase coverage"